code/parse/sexp.cpp

```diff
@@ -14742,6 +14756,73 @@ void sexp_send_message(int n)
 	sexp_send_one_message( name, who_from, priority, 0, 0, -1, do_hash_fallback );
 }
 
+ship* get_builtin_message_sender(const char* name) {
+	auto ship_entry = ship_registry_get(name);
+	if (ship_entry && ship_entry->has_shipp()) {
+		return ship_entry->shipp();
+	}
+	
+	auto wing_index = wing_lookup(name);
+	if (wing_index >= 0) {
+		auto ship_index = ship_get_random_ship_in_wing(wing_index, SHIP_GET_UNSILENCED);
+		return ship_index < 0 ? nullptr : &Ships[ship_index];
+	}
+
+	if (!stricmp(name, "<any wingman>")) {
+		auto ship_index = ship_get_random_player_wing_ship(SHIP_GET_UNSILENCED);
+		return ship_index < 0 ? nullptr : &Ships[ship_index];
+	}
+
+	return nullptr;
+}
+
+void sexp_send_builtin_message(int n) {
+	// Argument 1: Message type
+	auto type = get_builtin_message_type(CTEXT(n));
+	n = CDR(n);
+	// Argument 2: Subject (or <none>)
+	auto subject_name = CTEXT(n);
+	auto need_subject = stricmp(subject_name, SEXP_NONE_STRING);
```

Comment:
We should explicitly make a comparison here or cast to bool, otherwise this'll get int as type, which may behave unintuitively for anyone working on this code.
```suggestion
	auto need_subject = stricmp(subject_name, SEXP_NONE_STRING) != 0;
```